src/rashdf/geom.py

```diff
@@ -488,3 +574,171 @@ def ineffective_points(self) -> GeoDataFrame:  # noqa D102
+                part_start : part_start + part_cnt
+            ]
+            elevations.append(xzdata)
+
+        xs_elev_df = xs_df[
+            ["xs_id", "River", "Reach", "RS", "Left Bank", "Right Bank"]
+        ].copy()
+        xs_elev_df["Left Bank"] = xs_elev_df["Left Bank"].round(2).astype(str)
+        xs_elev_df["Right Bank"] = xs_elev_df["Right Bank"].round(2).astype(str)
+        xs_elev_df["elevation info"] = elevations
+
+        return xs_elev_df
+
+    def cross_sections_encroachment_station_left(self) -> pd.DataFrame:
+        """Return the left side encroachment information for a floodway plan hdf.
+
+        Returns
+        -------
+        DataFrame
+            A DataFrame containing the left side encroachment stations
+        """
+        path = "/Results/Steady/Output/Output Blocks/Base Output/Steady Profiles"
+        path += "/Cross Sections/Additional Variables/Encroachment Station Left"
+        if path not in self:
+            return pd.DataFrame()
+
+        profiles = self.steady_flow_names()
+
+        enc_data = self[path]
+        df_enc = pd.DataFrame(enc_data, index=profiles)
+        df_enc_t = df_enc.T.copy()
+        for p in profiles:
+            df_enc_t[p] = df_enc_t[p].apply(lambda x: round(x, 2))
+
+        return df_enc_t
+
+    def cross_sections_encroachment_station_right(self) -> pd.DataFrame:
+        """Return the right side encroachment information for a floodway plan hdf.
+
+        Returns
+        -------
+        DataFrame
+            A DataFrame containing the right side encroachment stations
+        """
+        path = "/Results/Steady/Output/Output Blocks/Base Output/Steady Profiles"
+        path += "/Cross Sections/Additional Variables/Encroachment Station Right"
+        if path not in self:
+            return pd.DataFrame()
+
+        profiles = self.steady_flow_names()
+
+        enc_data = self[path]
+        df_enc = pd.DataFrame(enc_data, index=profiles)
+        df_enc_t = df_enc.T.copy()
+        for p in profiles:
+            df_enc_t[p] = df_enc_t[p].apply(lambda x: round(x, 2))
+
+        return df_enc_t
+
+    def cross_sections_area(self) -> pd.DataFrame:
+        """Return the cross section area for each profile.
+
+        Returns
+        -------
+        DataFrame
+            A DataFrame containing the wet area inside the xs
+        """
+        path = "/Results/Steady/Output/Output Blocks/Base Output/Steady Profiles"
+        path += "/Cross Sections/Additional Variables/Area including Ineffective Total"
+        if path not in self:
+            return pd.DataFrame()
+
+        area_data = self[path]
+
+        profiles = self.steady_flow_names()
+
+        df_xsarea = pd.DataFrame(area_data, index=profiles)
+        df_xsarea_t = df_xsarea.T.copy()
+        for col in df_xsarea_t.columns:
+            df_xsarea_t[col] = df_xsarea_t[col].apply(lambda x: round(x, 2))
+
+        return df_xsarea_t
+
+    def cross_sections_velocity(self) -> pd.DataFrame:
```

Comment:
1. Methods pulling from `.p**.hdf` output files (i.e. `/Results/...` path) belong in `RasPlanHdf` rather than `RasGeomHdf`, since the latter is meant for opening `.g**.hdf` files.
3.  Noticing that these methods all have nearly identical code:
* `cross_sections_encroachment_station_left`
* `cross_sections_encroachment_station_right`
* `cross_sections_area`
* `cross_sections_velocity`

Could make this simpler by having a single private function to retrieve output data and an enum with the names of all the variables.

```python
class XsSteadyOutput(Enum):
    ALPHA = "Alpha"
    AREA_FLOW_CHANNEL = "Area Flow Channel"
    AREA_FLOW_LEFT_OB = "Area Flow Left OB"
    AREA_FLOW_RIGHT_OB = "Area Flow Right OB"
    ...
    WETTED_PERIMETER_TOTAL = "Wetted Perimeter Total"

DEFAULT_XS_STEADY_OUTPUT = [
    XsSteadyOutputVar.ENERGY_GRADE,
    XsSteadyOutputVar.FLOW,
    XsSteadyOutputVar.WATER_SURFACE,
]

class RasPlanHdf(RasGeomHdf):
    ...
    def steady_profile_xs_output(self, var: XsSteadyOutputVar, round_to: int = 2) -> DataFrame
        if var in DEFAULT_XS_STEADY_OUTPUT:
            path = f"{self.STEADY_OUTPUT_XS}/{var.value}"
        else:
            path = f"{self.STEADY_OUTPUT_XS_ADDITIONAL_VARS}/{var.value}"
        if path not in self:
            return pd.DataFrame()

        profiles = self.steady_flow_names()

        enc_data = self[path]
        df = pd.DataFrame(enc_data, index=profiles)
        df_t = df.T.copy()
        for p in profiles:
            df_t[p] = df_t[p].apply(lambda x: round(x, round_to))

        return df_t
```

Now variable-specific methods can be added without code duplication:
```python
class RasPlanHdf(RasGeomHdf):
    ...
    def cross_sections_wsel(self) -> DataFrame:
        return self.steady_profile_xs_output(XsSteadyOutput.WATER_SURFACE)

    def cross_sections_flow(self) -> DataFrame:
        return self.steady_profile_xs_output(XsSteadyOutput.FLOW)

    def cross_sections_energy_grade(self) -> DataFrame:
        return self.steady_profile_xs_output(XsSteadyOutput.ENERGY_GRADE)

    def cross_sections_additional_output(self) -> DataFrame:
        df = None
        for var in XsSteadyOutput:
            if var not in DEFAULT_XS_STEADY_OUTPUT:
                df_output = self.steady_profile_xs_output(var).add_suffix(f" {var.value}")
                if df is None:
                    df = df_output
                else:
                    df = df.merge(df_output)  # not sure if this is the correct merge syntax
        return df
```

Reply:
☝ Not trying to be prescriptive, btw, just pointing out an existing strategy in the codebase. I'm all for better ways to avoid code duplication or strong arguments in favor of it for specific cases.

Reply:
I'm in favor of doing so as well. I appreciate the instruction and example. You should see the update in the plan file following the path classes and functions for the steady results. You're suggested draft was very helpful. I'd like to keep things simple and in the spirit of the other code written. Works fantastic too!